scalableminds/webknossos
# Properly reserve dataset when putting datasource (addRemote)

Datastore now calls `reserveDataSourceUpload` when adding a datasource. This sets both uploader and folderId properly, as well as performs the correct write access checks.

(Note that this is a subset of #7176 as I expect that to be blocked for a while, but we should fix this bug. I copied the changes to the two backend source files from that PR)

### Steps to test:
- Add remote dataset as non-admin user, see that this works for folders you have write access for (and not for others)
- Check that the folder is set correctly. (Also, uploader should be set in db)

### TODOs:
- [x] Frontend: pass folderId to the put route *instead of* calling updatePartial afterwards.

### Issues:
- fixes #7215
- fixes #7211
- fixes #7199

------
